keps/sig-node/4328-node-label-manager/README.md

```diff
@@ -0,0 +1,442 @@
+-->
+
+##### Unit tests
+
+- Test that the validation returns an error when given an invalid NodeSelector.
+- Test that the validation pass when given a valid NodeSelector.
+- Test that the node_affinity plugin in scheduler handles the new type correctly.
+- Test that the NodeAffinityController receive the pod/node events, and handle them correctly(delete events occurred if pod's affinity doesn't match with node anymore). 
+
+##### Integration tests
+- Test that pods with `RequiredDuringSchedulingRequiredDuringExecution` node affinity are correctly scheduled.
+- Test that `RequiredDuringSchedulingRequiredDuringExecution` works well with other scheduler plugins.
+
+##### e2e tests
+- Test that pods that don't satisfy `RequiredDuringSchedulingRequiredDuringExecution` rule are correctly evicted.
+- Test that pods with `RequiredDuringSchedulingRequiredDuringExecution` are scheduled to correct nodes. 
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial unit/e2e tests completed and enabled
+- Documentation is added to demonstrate why this is useful how exactly affinity needs to be configured.
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Additional e2e tests are completed(if needed)
+
+#### GA
+
+- No negative feedback
+- No bug issues reported
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+The previous node affinity will not be broken. Users can continue to use their affinity as it is.
+To use this enhancement, users need to enable the feature gate, and add `RequiredDuringSchedulingRequiredDuringExecution` field for pod's node affinity.
+
+#### Downgrade
+The `RequiredDuringSchedulingRequiredDuringExecution` field will be ignored during both the scheduling phase and the execution phase. 
+
+### Version Skew Strategy
+
+kube-controller-manager, kube-scheduler and kube-apiserver will need enable the feature gate for the full featureset
+to be present. 
+
+If kube-apiserver doesn't enable the feature gate, this new affinity will be ignored.
+
+If kube-apiserver enable the feature gate, and:
+if only the kube-controller-manager enables the feature gate, the node affinity will be ignored during the scheduling phase.
+if only the kube-scheduler enables the feature gate, the node affinity will be ignored during the execution phase. 
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: NodeAffinityController 
+  - Components depending on the feature gate: kube-controller-manager, kube-scheduler, kube-apiserver
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
```

Comment:
I think answers are missing here?

Reply:
This feature will be controlled under a feature-gate. 
And I think these two questions are for those features which do not use a feature-gate to control? 😵‍💫

Reply:
you are supposed to remove the options that don't apply.